src/builddef.c

```diff
@@ -109,30 +109,34 @@ int main(int argc, char **argv) {
 
             char *bootdef_value;
 
-            if ( ( bootdef_value = strtok(NULL, "\n") ) != NULL ) {
-
+            bootdef_value = strtok(NULL, "\n");
+            char empty[] = "";
+            if (bootdef_value == NULL) {
+                bootdef_value = empty;
+            } else {
                 chomp_comments(bootdef_value);
-                singularity_message(VERBOSE2, "Got bootstrap definition key/val '%s' = '%s'\n", bootdef_key, bootdef_value);
-
-                if ( envar_defined(strjoin("SINGULARITY_DEFFILE_", uppercase(bootdef_key))) == 0 ) {
-                    singularity_message(ERROR, "Duplicate bootstrap definition key found: '%s'\n", bootdef_key);
-                    ABORT(255);
-                }
-
-                if ( strcasecmp(bootdef_key, "import") == 0 ) {
-                    // Do this again for an imported deffile
-                    bootstrap_keyval_parse(bootdef_value);
-                }
-
-                if ( strcasecmp(bootdef_key, "bootstrap") == 0 ) {
-                    singularity_registry_set("DRIVER", bootdef_value);
-                }
-
-                // Cool little feature, every key defined in def file is transposed
-                // to environment
-                envar_set(uppercase(bootdef_key), bootdef_value, 1);
-                envar_set(strjoin("SINGULARITY_DEFFILE_", uppercase(bootdef_key)), bootdef_value, 1);
             }
+
+            singularity_message(VERBOSE2, "Got bootstrap definition key/val '%s' = '%s'\n", bootdef_key, bootdef_value);
+
```

Comment:
We need to be careful here in the case that the user has put some kind of secret in the recipe (definitely not good practice or recommended, but it could happen!) Is there any way to filter / guarantee this? For the python bit I know the variables in advance so I just quiet them, but this seems like a catch all sort of deal.

Reply:
I agree on the need to be careful, that said, I didn't write this line of code, this was here before as well. I just unindented a whole block of code after changing the condition above. I can change this, but this was here the whole time even before my changes.

Reply:
ahhh, need to pull out the Github blame @GodloveD @gmkurtzer @bauerm97 !! :P